src/main/java/org/broadinstitute/hellbender/engine/MultiVariantWalkerBase.java

```diff
@@ -0,0 +1,99 @@
+package org.broadinstitute.hellbender.engine;
+
+import htsjdk.samtools.SAMSequenceDictionary;
+import htsjdk.variant.variantcontext.VariantContext;
+import htsjdk.variant.vcf.VCFHeader;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.SortedSet;
+import java.util.Spliterator;
+
+public abstract class MultiVariantWalkerBase extends VariantWalkerBase {
```

Comment:
If the only reason for factoring this out as a base class is to allow subclasses to specify a particular command line argument name for inputs, I think it would make more sense to use an ArgumentCollection and overridable method. Take a look at how GATKTool [uses that pattern](https://github.com/broadinstitute/gatk/blob/9be98b976815e6ca68d2e786139a48f0f21d8f77/src/main/java/org/broadinstitute/hellbender/engine/GATKTool.java#L51-L58) to allow tool subclasses to control the optional/required attribute on an argument. Just add an overridable method with a default implementation that returns an instance of a class that contains the `@Argument` definition, and has a method to return the driving variants. It will require addition of a new interface or base class for the arg collection, but I think thats a cleaner solution than introducing a new level into the class hierarchy. Of course that won't work if there are other reasons for factoring out the base class.

Reply:
I would say for now to confine it to the MultiVariantWalker hierarchy unless that can't be done cleanly.

Reply:
ok, done